core/src/providers/google_ai_studio.rs

```diff
@@ -446,32 +525,64 @@ impl LLM for GoogleAiStudioLLM {
         )
         .await?;
 
+        let mut content: Option<String> = None;
+        let mut function_calls: Vec<ChatFunctionCall> = vec![];
+
+        match c.candidates {
+            None => (),
+            Some(candidates) => match candidates.len() {
+                0 => (),
+                _ => match &candidates[0].content.parts {
+                    None => (),
+                    Some(parts) => {
+                        for p in parts.iter() {
```

Comment:
Not sure to get the value of using a for loop here 🙃

Reply:
simpler error reporting. Otherwise I need to return a Result and as a consequence use a map etc...

Reply:
Would probably be more readable, more idiomatic, and shorter though